examples/with-apollo-auth/lib/withData.js

```diff
@@ -46,28 +47,34 @@ export default ComposedComponent => {
         }
 
         // Provide the `url` prop data in case a graphql query uses it
-        const url = {query: context.query, pathname: context.pathname}
-
-        // Run all graphql queries
-        const app = (
-          <ApolloProvider client={apollo}>
-            <ComposedComponent url={url} {...composedInitialProps} />
-          </ApolloProvider>
-        )
-        await getDataFromTree(app, {
-          router: {
-            query: context.query,
-            pathname: context.pathname,
-            asPath: context.asPath
-          }
-        })
+        const url = { query: context.query, pathname: context.pathname }
+        try {
+          // Run all GraphQL queries
+          const app = (
+            <ApolloProvider client={apollo}>
+              <ComposedComponent url={url} {...composedInitialProps} />
+            </ApolloProvider>
+          )
+          await getDataFromTree(app, {
+            router: {
+              query: context.query,
+              pathname: context.pathname,
+              asPath: context.asPath
+            }
+          })
+        } catch (error) {
+          // Prevent Apollo Client GraphQL errors from crashing SSR.
+          // Handle them in components via the data.error prop:
+          // http://dev.apollodata.com/react/api-queries.html#graphql-query-data-error
+        }
+        // getDataFromTree does not call componentWillUnmount
+        // head side effect therefore need to be cleared manually
+        Head.rewind()
 
         // Extract query data from the Apollo's store
-        const state = apollo.getInitialState()
-
         serverState = {
-          apollo: { // Make sure to only include Apollo's data state
-            data: state.data
+          apollo: {
+            data: apollo.cache.extract()
```

Comment:
Can be simplified to just:

```js
serverState = apollo.cache.extract();
```
The change in the constructor can then be reverted:

```diff
-     this.apollo = initApollo(this.props.serverState.data, {
+     this.apollo = initApollo(this.props.serverState, {
```